tuning/sft_trainer.py

```diff
@@ -339,16 +341,40 @@ def train(
     }
     training_args = SFTConfig(**transformer_kwargs, **additional_args)
 
-    trainer = SFTTrainer(
-        model=model,
-        tokenizer=tokenizer,
-        train_dataset=formatted_train_dataset,
-        eval_dataset=formatted_validation_dataset,
-        data_collator=data_collator,
-        args=training_args,
-        callbacks=trainer_callbacks,
-        peft_config=peft_config,
-    )
+    trainer = None
+    if train_args.enable_reduce_loss_sum:
+        trainer = SumLossSFTTrainer(
```

Comment:
If the parameters are same across both, would it make sense to have a variable to capture the class to use, and do the initialisation call only once?

Reply:
you would have to initialize a class anyway to know if the model is PEFT or not 
https://github.com/foundation-model-stack/fms-hf-tuning/pull/520/files#diff-7d8f6f8626340337d03a2f04fec2840d2ede32b0c7af2640eed9759af45bb3cfR357